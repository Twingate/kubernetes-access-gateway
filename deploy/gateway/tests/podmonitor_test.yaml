suite: Pod Monitor
templates:
  - podmonitor.yaml
tests:
  - it: should enable Prometheus Pod Monitor
    set:
      metrics:
        podMonitor:
          enabled: true
    asserts:
      - matchSnapshot: {}
