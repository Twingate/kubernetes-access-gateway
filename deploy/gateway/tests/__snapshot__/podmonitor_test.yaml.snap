should enable Prometheus Pod Monitor:
  1: |
    apiVersion: monitoring.coreos.com/v1
    kind: PodMonitor
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kubernetes-access-gateway
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: kubernetes-access-gateway-0.1.0
      name: RELEASE-NAME-kubernetes-access-gateway
    spec:
      podMetricsEndpoints:
        - interval: 30s
          port: metrics
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kubernetes-access-gateway
